frontend/elements/src/entry/jig/publish/dropdown-list-object.ts

```diff
@@ -0,0 +1,58 @@
+import { MEDIA_UI } from '@utils/path';
+import { LitElement, html, css, customElement, property } from 'lit-element';
+
+@customElement('dropdown-list-object')
+export class _ extends LitElement {
+
+  static get styles() {
+    return [css`
+    p{
+    
+        margin-left:12px;
+        margin-top:0;
+        margin-bottom:0;
+  
+    }
+    .wrapper{
+        display:flex;
+        align-items:center;
+        margin-top:16px;
+        justify-content:space-between;
+        padding-right:16px;
+        position:relative;
+    }
+    .inside-wrapper{
+        display:flex;
+        align-items:center;
+    }
+    ::slotted([slot="tooltip"]){
+        position:absolute;
+        left: 110px;
+    top: -30px;
+    }
+   
+    `];
+  }
+@property()
+label: string = "";
+@property()
+icon: string = "";
+
+  
+
+  render() {
+    const {icon, label} = this;
+   
+   
+
+    return html`
+<div class="wrapper">
+    <div class="inside-wrapper">
+        <img-ui path="${icon}"></img-ui>
```

Comment:
generally speaking, UI images are predefined and should be a variant or hardcoded - e.g. it doesn't make sense to accept any image in our dropbox as a possibility here.

Reply:
If you look at publish-two component you can see how it is being used. Would that be an element in itself?

Reply:
The file as it is here is under `publish` so I assume this `dropdown-list-object` is used _only_ on the publish page. If that's the case, `Icon` should be a variant with those options However it seems these icons are also paired with hardcoded strings, so it's not really about the icon, it's about the overall mode, so `type Mode = "share" | "embed" | "link"`

Then, inside of the render function, you derive the icon and string based on the mode. So you ultimately have something like:

```
`export type Mode = "share" | "embed" | "link"

@customElement('dropdown-list-object')
export class _ extends LitElement {
  @property()
  mode: Mode = "share"
  
  render() {
    const {mode} = this;
    
    const iconPath = mode === "share" ? "share.svg"
      : mode === "embed" ? "embed.svg"
      : "link.svg";
      
     const label = mode === "share" ? SHARE_STR
      : mode === "embed" ? EMBED_STR
      : LINK_STR;
      
      return html`
         <p>${label}</p>
         <img-ui path="${iconPath}"></img-ui>
      `;
        
  }
}
```

Sidenote - "share" seems more appropriate than "student" which is why I called it that here

Reply:
If `dropdown-list-object` is more generic and used everywhere then I'd divide it into two separate things:

1. A generic element that only has a slot for the image. Idea is it can be used for _any_ image - not just ui
2. This element which works the same way as described with mode, but doesn't have all the HTML/CSS, rather it merely renders the generic element and supplies the label/icon (with the icon being placed in the slot)

Reply:
I will change it into mode